src/suricata.c

```diff
@@ -262,17 +262,30 @@ void EngineModeSetIDS(void)
 
 int RunmodeIsUnittests(void)
 {
-    if (run_mode == RUNMODE_UNITTEST)
+    if (runmodeslist.run_mode[0] == RUNMODE_UNITTEST)
         return 1;
 
     return 0;
 }
 
-int RunmodeGetCurrent(void)
+int RunmodeGetCurrent(int index)
 {
-    return run_mode;
+    return runmodeslist.run_mode[index];
 }
 
+static int RunmodeIsAlreadyRan(RunModesList *runmodes, int run_mode)
```

Comment:
int run_mode should be enum RunModes? Also, what is the purpose of this func? Add comment to it plz.